mir_eval/sonify.py

```diff
@@ -128,21 +139,41 @@ def _fast_synthesize(frequency):
         # Use a flatiter to simulate a long 1D buffer
         return long_signal.flat
 
+    def _const_interpolator(value):
+        """Return a function that returns `value`
+            no matter the input.
+        """
+        def __interpolator(x):
+            return value
+        return __interpolator
+
     # Threshold the tfgram to remove non-positive values
     gram = np.maximum(gram, 0)
 
     # Pre-allocate output signal
     output = np.zeros(length)
+    time_centers = np.mean(times, axis=1) * float(fs)
+
     for n, frequency in enumerate(frequencies):
         # Get a waveform of length samples at this frequency
         wave = _fast_synthesize(frequency)
+        if len(time_centers) > 1:
+            gram_interpolator = interp1d(
+                time_centers, gram[n, :], kind='linear',
+                bounds_error=False, fill_value=0.0)
+        else:
+            # if there is only one point in time_centers, interpolator
+            # returns constant
+            gram_interpolator = _const_interpolator(gram[n, 0])
```

Comment:
Can you replace all this logic with 

```Python
gram_interpolator = interp1d(
    time_centers, gram[n, :],
    kind='linear' if len(time_centers) > 1 else 'nearest',
    bounds_error=False, fill_value=0.0)
```
or maybe `'zero'` instead of `'nearest'`?  Alternatively if you can provide a little more of a comment for special casing, would be helpful for understanding it.